test/assert.js

```diff
@@ -530,6 +530,65 @@ describe('assert', function () {
     }, "blah: expected {} to be a function");
   });
 
+  it('isCallable', function() {
+    var func = function() {};
+    assert.isCallable(func);
+
+    var func = async function() {};
+    assert.isCallable(func);
+
+    var func = function* () {}
+    assert.isCallable(func);
+
+    var func = async function* () {}
+    assert.isCallable(func);
+
+    err(function () {
+      assert.isCallable({}, 'blah');
+    }, "blah: expected {} to be callable");
+  });
+
+  it('isAsyncFunction', function() {
+    var func = async function() {};
+    assert.isAsyncFunction(func);
+
+    var func = async function*() {};
+    assert.isAsyncFunction(func);
+
+    err(function () {
+      assert.isAsyncFunction(function() {}, 'blah');
+    }, "blah: expected [Function] to be a AsyncFunction");
+  });
+
+  it('isGeneratorFunction', function() {
+    var func = function* () {}
+    assert.isGeneratorFunction(func) 
+
+    var func = async function* () {}
+    assert.isGeneratorFunction(func) 
+
+    err(function () {
+      assert.isGeneratorFunction(function() {}, 'blah');
+    }, "blah: expected [Function] to be a GeneratorFunction");
+  })
+  
+  it('isAsyncGeneratorFunction', function() {
+    var func = async function* () {}
+    assert.isAsyncGeneratorFunction(func) 
+
+    err(function () {
+      assert.isAsyncGeneratorFunction(function() {}, 'blah');
+    }, "blah: expected [Function] to be a AsyncGeneratorFunction");
+    
+    err(function () {
+      assert.isAsyncGeneratorFunction(async function() {}, 'blah');
+    }, "blah: expected [Function] to be a AsyncGeneratorFunction");
```

Comment:
This should probably be:

```suggestion
    }, "blah: expected [AsyncFunction] to be a AsyncGeneratorFunction");
```

but needs a change in `loupe`. See https://github.com/chaijs/loupe/pull/77